scipy/io/matlab/mio5_utils.pyx

```diff
@@ -914,7 +914,7 @@ cdef class VarReader5:
             int *n_duplicates
             char **name_ptrs
         n_duplicates = <int *>calloc(n_names, sizeof(int))
-        name_ptrs = <char **>calloc(n_names, sizeof(char *))
+        name_ptrs = <char **>malloc(n_names * sizeof(char *))
```

Comment:
This doesn't even need to be allocated at all. You can change it to
```cython
cdef char * names_ptr = names
```
Then remove all other references except line 927 which should be changed to
```cython
if strcmp(n_ptr, names_ptr[j * namelength]) == 0: # the same
```

Reply:
Great point! I have made the changes you suggested.